app/controllers/comfy/admin/api_resources_controller.rb

```diff
@@ -31,18 +29,12 @@ def create
     respond_to do |format|
       if @api_resource.save
         load_api_actions_from_api_resource
-        handle_custom_actions if @custom_actions.present?
-        handle_serve_file_action if @serve_file_action.present?
-        # TO DO: this needs to be handled as well
-        # handle_redirection if @redirect_action.present?
+        execute_api_actions
 
-        format.html {
-          if @redirect_action.present?
-            handle_redirection
-          else
-            redirect_to api_namespace_resource_path(api_namespace_id: @api_resource.api_namespace_id,id: @api_resource.id), notice: "Api resource was successfully created."
-          end
-        }
+        # Preventing double render error
+        return if @redirect_action.present?
```

Comment:
Since, we are executing the api_actions in a defined order, we need to prevent the double render error.